### PR TITLE
refactor(app): implement full cal check summary

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

Builds out the Summary View of the Robot Calibration Check completion step. 

NOTE: This does not yet add the functionality to download the JSON results as a file, that will come in a following PR.

## changelog

- Fill out CompleteConfirmation.js with newly designed summary
- Build two abstracted components (`DifferenceValue` and `ThresholdValue`) for formatting threshold and difference values that should be reused once the Comparison pages are updated.
- Pull out comparison table per pipette into new component called `PipetteComparisons` 

## review requests

- Run the calibration check and make sure the summary view present the correct data and matches the design.

## risk assessment

Low, mostly presentational logic, behind a feature flag

### CODE REVIEW

File: app/src/components/CheckCalibration/CompleteConfirmation.js [hunk 1]
```diff
@@ -1,125 +1,95 @@
...
-const DELETE_ROBOT_CALIBRATION_CHECK_HEADER =
-  'Robot calibration check is complete'
-const DELETE_ROBOT_CALIBRATION_CHECK_BUTTON_TEXT = 'Drop tip in trash and exit'
-const CHECK = 'check'
-const CHECKS = 'checks'
-const PASSED = 'passed'
-const FAILED = 'failed'
+const ROBOT_CALIBRATION_CHECK_SUMMARY_HEADER = 'Calibration check summary:'
+const DROP_TIP_AND_EXIT = 'Drop tip in trash and exit'
+const DOWNLOAD_SUMMARY = 'Download JSON summary'
```
Comment (sfoster1) [inline app/src/components/CheckCalibration/CompleteConfirmation.js:18]: nitpick - can we make this say 'copy JSON summary to clipboard' for the interim until the download is actually implemented

**Review by nusrat813 — APPROVED** [review]
Tested - working as intended

File: app/src/components/CheckCalibration/CompleteConfirmation.js [hunk 2]
```diff
@@ -1,125 +1,95 @@
...
           type="text"
           value={JSON.stringify(comparisonsByStep)}
           onFocus={e => e.currentTarget.select()}
           readOnly
-        />
-        <IconButton
-          className={styles.copy_icon}
-          onClick={handleCopyButtonClick}
-          name="ot-copy-text"
+          style={{ opacity: 0 }}
```
Comment (mcous) [inline app/src/components/CheckCalibration/CompleteConfirmation.js:83]: If the copy button still works, I think `type="hidden"` might be more semantically correct than hiding via CSS

File: app/src/components/CheckCalibration/__tests__/DifferenceValue.test.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        <DifferenceValue
+          stepName={stepName}
+          differenceVector={differenceVector}
+        />
+      )
+    }
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
```
Comment (mcous) [inline app/src/components/CheckCalibration/__tests__/DifferenceValue.test.js:37]: I don't see any mocks in this file, is that intentional?

**Review by mcous — COMMENTED** [review]
About to jump into testing but here's my code review

File: app/src/components/CheckCalibration/CompleteConfirmation.js (hunk 2 above)
Comment (b-cooper) [reply app/src/components/CheckCalibration/CompleteConfirmation.js:83]: I tried the hidden prop first, but it breaks the copying functionality. This is going to be immediately replaced by a PR that Seth is working on, so I'm not worried about it.

File: app/src/components/CheckCalibration/__tests__/DifferenceValue.test.js (hunk 3 above)
Comment (b-cooper) [reply app/src/components/CheckCalibration/__tests__/DifferenceValue.test.js:37]: I guess I deleted them all... 👍

File: app/src/components/CheckCalibration/CompleteConfirmation.js (hunk 2 above)
Comment (mcous) [reply app/src/components/CheckCalibration/CompleteConfirmation.js:83]: Copy to clipboard **does not** work with type="hidden"

**Review by mcous — APPROVED** [review]
LGTM. Was unable to pick up a tip without it yelling at me, so I faked the tip pickup to run through the flow.

Summary page looked good

**Pull request merged**